packages/vite/src/node/preview.ts

```diff
@@ -83,6 +86,21 @@ export async function preview(
     'production',
   )
 
+  const distDir = path.resolve(config.root, config.build.outDir)
+  const isAlreadyBuild = fs.existsSync(distDir)
+  if (!isAlreadyBuild) {
+    const { status, error } = spawn.sync('npx', ['vite', 'build'])
```

Comment:
I don't think we should execute `vite build` within `preview`. It feels a bit too automatic, and doesn't work for metaframeworks that use `preview` API but has a different build command too.

I think for now it's better to leave it as-is. If we really want to check, we could see if any plugins implement `configurePreviewServer` before issuing the warning. We might also want to do that if it's running through the `vite preview` command, as using the `preview` API, the consumer could adjust `previewServer.httpServer` themselves.